src/pytest/test_replicate.py

```diff
@@ -1,23 +1,48 @@
 import subprocess
+import signal
 import os
 import os.path
 from RLTest import Env
 import time
 
+class TimeLimit(object):
+    """
+    A context manager that fires a TimeExpired exception if it does not
+    return within the specified amount of time.
+    """
+    def __init__(self, timeout):
+        self.timeout = timeout
+    def __enter__(self):
+        signal.signal(signal.SIGALRM, self.handler)
+        signal.setitimer(signal.ITIMER_REAL, self.timeout, 0)
+    def __exit__(self, exc_type, exc_value, traceback):
+        signal.setitimer(signal.ITIMER_REAL, 0)
+        signal.signal(signal.SIGALRM, signal.SIG_DFL)
+    def handler(self, signum, frame):
+        raise Exception()   
+
 def testDelReplicate():
   env = Env(useSlaves=True)
   master = env.getConnection()
-  slave = env.getSlaveConnection()  
+  slave = env.getSlaveConnection() 
+
   env.assertContains("PONG", master.execute_command("ping"))
   env.assertContains("PONG", slave.execute_command("ping"))  
-
   env.assertOk(master.execute_command('ft.create', 'idx', 'schema', 'f', 'text'))
 
   for i in range(10):
     master.execute_command('ft.add', 'idx', 'doc%d' % i, 1.0, 'fields',
                                       'f', 'hello world')
-
-  time.sleep(5)
+  # Ensure slave is updated
+  master.execute_command('set foo bar')
+  try:
```

Comment:
I don't see why we need `try` here; it should just fail..

Reply:
I got the code from @MeirShpilraien.
I believe this will give a cleaner error message on the test.

Reply:
Mark I believe that a general timeout code is better then checking the time each iteration. Code looks cleaner.